tests/src/test/scala/ch/epfl/bluebrain/nexus/tests/kg/ResourcesSpec.scala

```diff
@@ -113,6 +113,19 @@ class ResourcesSpec extends BaseSpec with EitherValuable with CirceEq {
         )
       }
     }
+
+    "fail if the id is blank" in {
+      deltaClient
+        .post[Json](s"/resources/$id1/_/", jsonContentOf("/kg/resources/simple-resource-with-blank-id.json"), Rick) {
+          (json, response) =>
+            response.status shouldEqual StatusCodes.BadRequest
+            json shouldEqual jsonContentOf(
+              "/kg/resources/invalid-resource-id.json",
+              "id" -> ""
+            )
+        }
+    }
```

Comment:
Arguably there should be a second test here, however I could not get the route to fire when specifying the id in the url. The best we could do is have a test which says that a resource is not created. This would pass because the route is rejected. Given the issue we are dealing with, it doesn't seem useful to have this test